common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/TakeActions.kt

```diff
@@ -156,6 +180,67 @@ class TakeActions @Inject constructor(
         }
     }
 
+    private fun getChunkAudioRange(sort: Int, max: Int, cues: List<AudioCue>): Pair<Int, Int> {
+        val current = cues[sort - 1].location
+        val next = if (sort in 0..cues.lastIndex) {
+            cues[sort].location
+        } else {
+            max
+        }
+
+        return Pair(current, next)
+    }
+
+    private fun createChunkFromChapterAudio(
+        chunk: Recordable,
+        takeFile: File,
+        chapterFile: File?
+    ): Pair<Take, Result> {
```

Comment:
Why not have this be a file that takes a chapter audio and creates a map of cue labels to PCMFile? (SplitAudioOnCues would probably be a good name for the use case class or function, as part of the common.audio package)

Rather than give it one chunk at a time, have it find the chunk, seek through the file, and extract out that chunk, for all chunks (an O(N^2) algorithm), you could get the metadata, scan over the file once and break into chunks as you go (an O(N) algorithm)

Results don't really make sense as a return type here, so missing chunks would just not have an entry into the map.

Creating a Take should be done probably as a second step, in trying to keep with the single responsibility, each function should do one thing, mentality as Take introduces a different domain concept (one being focused on specific audio file processing, the other introducing our application domain jargon)